datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -369,7 +356,7 @@ impl ExternalSorter {
                 .with_metrics(self.metrics.baseline.clone())
                 .with_batch_size(self.batch_size)
                 .with_fetch(self.fetch)
-                .with_reservation(self.reservation.new_empty())
+                .with_reservation(self.merge_reservation.new_empty())
```

Comment:
I believe we should use `merge_reservation` here, because the allocations in the stream built here are for merging.